src/sync.cpp

```diff
@@ -168,44 +190,36 @@ namespace librealsense
             if (dev)
             {
                 dev_exist = true;
-                matcher = _matchers[stream_id];
-                if (!matcher)
-                {
-                    std::ostringstream ss;
-                    for (auto const & it : _matchers)
-                        ss << ' ' << it.first;
-                    LOG_DEBUG("stream id " << stream_id << " was not found; trying to create, existing streams=" << ss.str());
-                    matcher = dev->create_matcher(frame);
-
-                    matcher->set_callback(
-                        [&](frame_holder f, syncronization_environment env)
-                    {
-                        sync(std::move(f), env);
-                    });
+                matcher = dev->create_matcher(frame);
```

Comment:
you removed the check if (!matcher) it means that you will create a new matcher every frame , its supposed to be only on the first frame.

Reply:
I didn't remove the check -- it's at the start of the function, no?